core/src/main/java/org/geysermc/geyser/session/GeyserSession.java

```diff
@@ -1484,4 +1486,36 @@ public void refreshEmotes(List<UUID> emotes) {
             player.sendUpstreamPacket(emoteList);
         }
     }
+
+    /**
+     * Send the following fog IDs, as well as the cached ones, to the client.
+     *
+     * Fog IDs can be found here:
+     * https://wiki.bedrock.dev/documentation/fog-ids.html
+     *
+     * @param fogNameSpaces the fog ids to add
+     */
+    public void sendFog(String... fogNameSpaces) {
+        this.fogNameSpaces.addAll(Arrays.asList(fogNameSpaces));
+
+        PlayerFogPacket packet = new PlayerFogPacket();
+        Collections.addAll(packet.getFogStack(), this.fogNameSpaces.toArray(new String[0]));
+        sendUpstreamPacket(packet);
+    }
+
+    /**
+     * Removes the following fog IDs from the client and the cache.
+     *
+     * @param fogNameSpaces the fog ids to remove
+     */
+    public void removeFog(String... fogNameSpaces) {
+        if (fogNameSpaces.length == 0) {
+            this.fogNameSpaces.clear();
+        } else {
+            this.fogNameSpaces.removeAll(Arrays.asList(fogNameSpaces));
+        }
+        PlayerFogPacket packet = new PlayerFogPacket();
+        Collections.addAll(packet.getFogStack(), this.fogNameSpaces.toArray(new String[0]));
```

Comment:
Can't you just do `packet.getFogStack().addAll(this.fogNameSpaces)` ? Same with the add method.

Reply:
I copied what was in `WorldBorder#sendFog` but ill change it